### ISSUE TITLE
feature: Link to function parameters

### ISSUE THREAD

**Issue by Kludex** [mkdocstrings/mkdocstrings#640]
### Is your feature request related to a problem? Please describe.

At the moment, I'm not able to create a link to parameters.

### Describe the solution you'd like

I would like to be able to create links to function parameters.

### Describe alternatives you've considered

My links are to the function, or some other structure that has the parameter names and a description, so I can document it - the first is the one I usually use, but not optimal, since the user has still to find out the parameter I'm talking about.

**Comment by samuelcolvin** [comment]
This would be really helpful for us @pawamoy, but I'm sure it would help many others.

Any chance it could e prioritied?

**Comment by pawamoy** [comment]
Sure, I'll up the priority. Do you want parameters to appear in the ToC?

**Comment by samuelcolvin** [comment]
Thanks so much.

I don't think that matters to us much, I think on balance, no but I can also imagine it might make more sense for them to appear. @Kludex WDYT?

**Comment by Kludex** [comment]
I think we are only interested on the links. If the ToC is more work, then no need.

**Comment by pawamoy** [comment]
It's actually the opposite, we can easily create  a heading for each parameter, and they'll naturally appear in the ToC. Hiding them from the ToC will be more work (already have an idea how, just not sure how well it'll work) :)

Another thing I'm not sure yet is how well this will play with section formats like table and spacy. I'll experiment a bit and report back.

**Comment by Kludex** [comment]
Ah, then ToC is good. 👀

**Comment by pawamoy** [comment]
Here's how it could look:

Dark ToC | Light ToC
--- | ---
![toc_dark](https://github.com/mkdocstrings/mkdocstrings/assets/3999221/eaebfbc8-6df6-4edb-a189-0f08b5a02f71) | ![toc_light](https://github.com/mkdocstrings/mkdocstrings/assets/3999221/c1b48aa8-1692-4361-999f-de843dc6b25f)

I'm not sold on the colors. It's a too bright pink. I'll try to make it less bright. Any other color suggestion (especially hex values :wink:) is welcome.

Section style | Preview
List | ![list](https://github.com/mkdocstrings/mkdocstrings/assets/3999221/ee7f247a-c588-4117-a154-8a39383364e3)
Spacy | ![spacy](https://github.com/mkdocstrings/mkdocstrings/assets/3999221/064cbf8b-830b-437e-86ce-a76b010faca4)
Table | ![table](https://github.com/mkdocstrings/mkdocstrings/assets/3999221/095ee91b-112d-4f3d-a3fd-f77a6bcbd4b9)

I don't think I'll keep the symbol types before parameter names. It's a bit redundant and adds too much visual noise. What do you think?

**Comment by Kludex** [comment]
On the ToC the symbol types look good, on the list itself I don't think so.

Thanks for this. 🙏

**Comment by pawamoy** [comment]
Another question: with such headings for parameters, they now have permalinks (the lime ¤ in screenshots above), and also are included in the generated objects inventory. What should their identifier be? Possible options:

- `path.to.function-param`
- `path.to.function(param)`
- any other suggestion?

I checked and it's not natively supported by Sphinx's interlink extension. There's a third-party extension supporting it (), and they went with `:paramref:`.EnvironmentContext.configure.transactional_ddl`. I disagree with this choice of separating the parameter name with a dot, because it could clash with a class member of the same name. Example:

```python
class A:
    def __init__(self, param):
        self.param = param  # clash

    # or
    def param(self):
        pass  # clash again
```

**Comment by Kludex** [comment]
I think the parenthesis will be faster to do the mental link, but I'm not strong here.

**Comment by pawamoy** [comment]
Thanks for the feedback! I'll go with `path.to.function(param)` then. Anyone reading this, please comment if you disagree with this syntax and/or have a better suggestion :slightly_smiling_face:

**Comment by samuelcolvin** [comment]
I'm the previews, I think all look great! Thank you.

On the linking, my guess is it doesn't matter that much as long as it doesn't break the markdown parser or any markdown highlighters - once the link is this complicated I guess most people will run the server and copy the actual link.

But I agree with @Kludex, parentheses look better if that's possible.

**Comment by pawamoy** [comment]
Good point about Markdown parsers. I'll make sure this syntax doesn't cause any issue :slightly_smiling_face:

**Comment by pawamoy** [comment]
This feature is now available in version 1.6.0 of mkdocstrings-python Insiders :slightly_smiling_face:
See the docs: https://mkdocstrings.github.io/python/usage/configuration/headings/#parameter_headings.
